src/data/complex/basic.lean

```diff
@@ -478,6 +481,127 @@ by rw [← of_real_int_cast, abs_of_real, int.cast_abs]
+  begin
+    obtain ⟨w₁, l₁, rfl⟩ := h₁,
+    obtain ⟨w₂, l₂, e⟩ := h₂,
+    have h₃ : w₁ + w₂ = 0,
+    { symmetry,
+      rw add_assoc at e,
+      apply of_real_inj.mp,
+      apply add_left_cancel,
+      convert e; simp, },
+    have h₄ : w₁ = 0, linarith,
+    simp [h₄],
+  end, }
+
+lemma le_def {z w : ℂ} : z ≤ w ↔ ∃ x : ℝ, 0 ≤ x ∧ w = z + x := iff.refl _
+lemma lt_def {z w : ℂ} : z < w ↔ ∃ x : ℝ, 0 < x ∧ w = z + x :=
+begin
+  rw [lt_iff_le_not_le],
+  fsplit,
+  { rintro ⟨⟨x, l, rfl⟩, h⟩,
+    by_cases hx : x = 0,
+    { simp [hx] at h, exfalso, exact h (le_refl _), },
+    { replace l : 0 < x := lt_of_le_of_ne l (by { symmetry, exact hx }),
+      exact ⟨x, l, rfl⟩, } },
+  { rintro ⟨x, l, rfl⟩,
+    fsplit,
+    { exact ⟨x, le_of_lt l, rfl⟩, },
+    { rintro ⟨x', l', e⟩,
+      rw [add_assoc] at e,
+      replace e := add_left_cancel (by { convert e, simp }),
+      norm_cast at e,
+      apply lt_irrefl (0 : ℝ),
+      conv { congr, skip, rw e, },
+      apply lt_of_lt_of_le l (le_add_of_nonneg_right l'), } }
+end
+
+@[simp, norm_cast] lemma real_le_real {x y : ℝ} : (x : ℂ) ≤ (y : ℂ) ↔ x ≤ y :=
+begin
+  rw [le_def],
+  fsplit,
+  { rintro ⟨r, l, e⟩,
+    norm_cast at e,
+    subst e,
+    exact le_add_of_nonneg_right l, },
+  { intro h,
+    refine ⟨y - x, sub_nonneg.mpr h, (by simp)⟩, },
+end
+@[simp, norm_cast] lemma real_lt_real {x y : ℝ} : (x : ℂ) < (y : ℂ) ↔ x < y :=
+begin
+  rw [lt_def],
+  fsplit,
+  { rintro ⟨r, l, e⟩,
+    norm_cast at e,
+    subst e,
+    exact lt_add_of_pos_right x l, },
+  { intro h,
+    refine ⟨y - x, sub_pos.mpr h, (by simp)⟩, },
+end
+@[simp, norm_cast] lemma zero_le_real {x : ℝ} : (0 : ℂ) ≤ (x : ℂ) ↔ 0 ≤ x := real_le_real
+@[simp, norm_cast] lemma zero_lt_real {x : ℝ} : (0 : ℂ) < (x : ℂ) ↔ 0 < x := real_lt_real
+
+/--
+With `z ≤ w` iff `w - z` is real and nonnegative, `ℂ` is an ordered ring.
+-/
+instance : ordered_comm_ring ℂ :=
+{ zero_le_one := ⟨1, zero_le_one, by simp⟩,
+  add_le_add_left := λ w z h y,
+  begin
+    obtain ⟨x, l, rfl⟩ := h,
+    refine ⟨x, l, by simp [add_assoc]⟩,
+  end,
+  mul_pos := λ z w hz hw,
+  begin
+    obtain ⟨zx, lz, rfl⟩ := lt_def.mp hz,
+    obtain ⟨wx, lw, rfl⟩ := lt_def.mp hw,
+    norm_cast,
+    simp only [mul_pos, lz, lw, zero_add],
+  end,
+  le_of_add_le_add_left := λ u v z h,
+  begin
+    obtain ⟨x, l, e⟩ := h,
+    rw add_assoc at e,
+    exact ⟨x, l, add_left_cancel e⟩,
+  end,
+  mul_lt_mul_of_pos_left := λ u v z h₁ h₂,
+  begin
+    obtain ⟨x₁, l₁, rfl⟩ := lt_def.mp h₁,
+    obtain ⟨x₂, l₂, rfl⟩ := lt_def.mp h₂,
+    simp only [mul_add, zero_add],
+    exact lt_def.mpr ⟨x₂ * x₁, mul_pos l₂ l₁, (by norm_cast)⟩,
+  end,
+  mul_lt_mul_of_pos_right := λ u v z h₁ h₂,
+  begin
+    obtain ⟨x₁, l₁, rfl⟩ := lt_def.mp h₁,
+    obtain ⟨x₂, l₂, rfl⟩ := lt_def.mp h₂,
+    simp only [add_mul, zero_add],
+    exact lt_def.mpr ⟨x₁ * x₂, mul_pos l₁ l₂, (by norm_cast)⟩,
+  end,
+  -- Why do we need these next four fields? They should be copied from `comm_ring ℂ`.
```

Comment:
I'd like to understand this too. Note that adding the following to the end works (though I don't think we should do this!):
```lean
  ..(by apply_instance : semiring ℂ), -- comm_semiring and even mul_zero_class work too!
  ..(by apply_instance : add_left_cancel_semigroup ℂ),
  ..(by apply_instance : add_right_cancel_semigroup ℂ),
```

Reply:
I've posted on Zulip to see if anyone can explain.